api/cmd/fetch/pipeline.go

```diff
@@ -36,7 +39,7 @@ type fragmentcache interface {
 
 type cacheEntry struct {
 	chunk []byte
-	etag  azblob.ETag
+	etag  *string
```

Comment:
I choose string pointer because that's how it's returned by Download [1], and that's how it's expected in ModifiedAccessConditions as well [2] . We could manually cast it back and forth it into `azcore.ETag`, but I don't see any gain from that. 

[1] https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/storage/azblob#BlobDownloadResult
[2] https://pkg.go.dev/github.com/Azure/azure-sdk-for-go/sdk/storage/azblob#ModifiedAccessConditions